File: algorithms/golang/1177-canMakePalindromeFromSubstring/Solution.go
package main

func canMakePaliQueries(s string, queries [][]int) []bool {
	// i 表示前i字符中每个字符出现的次数
	sum := make([][26]int, len(s)+1, len(s)+1)
	for i := 0; i < len(s); i++ {
		sum[i+1] = sum[i]
		sum[i+1][s[i]-'a']++
	}

	can := make([]bool, len(queries), len(queries))
	for i, query := range queries {
		diff := 0
		for j := 0; j < 26; j++ {
			diff += (sum[query[1]+1][j] - sum[query[0]][j]) % 2
		}

		can[i] = diff/2 <= query[2]
	}
	return can
}

File: algorithms/java/1177-canMakePalindromeFromSubstring/Solution.java
import java.util.List;
import java.util.ArrayList;

class Solution {
  public List<Boolean> canMakePaliQueries(String s, int[][] queries) {
    int[][] sum = new int[s.length()+1][];
    sum[0] = new int[26];
    for (int i = 0; i < s.length(); i++) {
      sum[i + 1] = sum[i].clone();
      ++sum[i + 1][s.charAt(i) - 'a'];
    }

    List<Boolean> can = new ArrayList<>(queries.length);
    for (int[] q: queries) {
      int diff = 0;
      for (int j = 0; j < 26; j++) {
        diff += (sum[q[1] + 1][j] - sum[q[0]][j]) & 1;
      }
      System.out.println(diff);
      can.add(diff/2<=q[2]);
    }
    return can;
  }
}